hip-config.cmake.in

```diff
@@ -153,24 +153,23 @@ endif()
 if(HIP_COMPILER STREQUAL "clang")
   if (HIP_CXX_COMPILER MATCHES ".*clang\\+\\+")
     set_property(TARGET hip::device APPEND PROPERTY
-      INTERFACE_COMPILE_OPTIONS -mllvm -amdgpu-early-inline-all=true -mllvm -amdgpu-function-calls=false
+      INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false>"
     )
   endif()
 
-  if (EXISTS ${AMD_DEVICE_LIBS_PREFIX}/amdgcn/bitcode)
-    set_property(TARGET hip::device APPEND PROPERTY
-      INTERFACE_COMPILE_OPTIONS -xhip
+  set_property(TARGET hip::device APPEND PROPERTY
+      INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-x hip>"
```

Comment:
I did another test (replicating the logic elsewhere) and this line was causing trouble unless I wrote it as:
```suggestion
      INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-xhip>"
```
With `"-x hip"`:
```
clang-12: error: language not recognized: ' hip'
```

Reply:
I wonder if doing `"$<$<COMPILE_LANGUAGE:CXX>:SHELL:-x> $<$<COMPILE_LANGUAGE:CXX>:SHELL:hip>"` would work as well.

Reply:
Or maybe `$<$<COMPILE_LANGUAGE:CXX>:SHELL:-x;hip>` like what is done for the other flags.

Reply:
The problem in my test was, that I had both `-xhip` and `"-x hip"` on my command line. Not entirely sure where the `""` came from, I am playing if I can find a work around until the PR herein lands:
```cmake
       target_compile_options(amrex PUBLIC
           "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false;-xhip>
```
If I added `...;-x hip` my CLI looked like this: "clang++ ... "-x hip" -xhip". Without the space it worked.
https://github.com/AMReX-Codes/amrex/pull/2031

```
$ clang++ -x hip
# ok
$ clang++ -xhip
# ok
$ clang++ "-x hip" -xhip
clang-12: error: language not recognized: ' hip'
$ clang++ "-x hip"
clang-12: error: language not recognized: ' hip'
$ clang++ -x hip
# ok
```

Anyway, I think that might just be from my work-around. The stand-alone test I did worked, as reported earlier.

Reply:
> Or maybe $<$<COMPILE_LANGUAGE:CXX>:SHELL:-x;hip> like what is done for the other flags.

That might be a bit risky, the idea of SHELL is to keep options that belong together also in lists together. The list `;` would tokenize them and potentially shuffle them later on.

Reply:
So I take this back, I think the current status is probably ok. We should trust on @robertmaynard's expertise in CMake things :D

Reply:
Thus I don't need update it .

@mangupta   will help us merge in weekend or next monday

Reply:
Confirmed. The problem was on my side, writing
```cmake
       target_compile_options(amrex PUBLIC
           "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false;-x hip>
```
where I should have written
```cmake
       target_compile_options(amrex PUBLIC
           "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-mllvm;-amdgpu-early-inline-all=true;-mllvm;-amdgpu-function-calls=false>
       target_compile_options(amrex PUBLIC
           "$<$<COMPILE_LANGUAGE:CXX>:SHELL:-x hip>
```